src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -883,6 +883,48 @@ class npc_ros_dark_rider : public CreatureScript
 
 };
 
+enum GiftOfTheHarvester
+{
+    SPELL_GIFT_OF_THE_HARVESTER = 52479,
+    SPELL_MINER_GHOUL_TRANSFORM = 52490,
+    SPELL_MINER_GHOST_TRANSFORM = 52505
+};
+
+struct gift_of_the_harvesterAI : public GameObjectAI
+{
+    gift_of_the_harvesterAI(GameObject* go) : GameObjectAI(go) { }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+
+        _scheduler.Schedule(2s, [this](TaskContext /*context*/)
```

Comment:
i dont like it.
You will reschedule this event in each tick of loop.
Move this schedule block to hooks like Reset()